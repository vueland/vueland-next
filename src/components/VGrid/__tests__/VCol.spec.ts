import { mount } from '@vue/test-utils'
import { VCol } from '../'

describe('VCol', () => {
  const mountFunction = (options = {}) =>
    mount(VCol, {
      ...options,
    })

  it('should test "grid" and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        xl: 12,
        lg: 4,
        md: 6,
        sm: 12,
      },
    })
    expect(wrapper.find('.xl-12').exists()).toBe(true)
    expect(wrapper.find('.lg-4').exists()).toBe(true)
    expect(wrapper.find('.md-6').exists()).toBe(true)
    expect(wrapper.find('.sm-12').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should test "grid order" and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        orderXl: 2,
        orderLg: 4,
        orderMd: 3,
        orderSm: 1,
      },
    })
    expect(wrapper.find('.order-xl-2').exists()).toBe(true)
    expect(wrapper.find('.order-lg-4').exists()).toBe(true)
    expect(wrapper.find('.order-md-3').exists()).toBe(true)
    expect(wrapper.find('.order-sm-1').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
