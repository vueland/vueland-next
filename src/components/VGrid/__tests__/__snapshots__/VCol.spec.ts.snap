// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECol should test "grid order" and match snapshot 1`] = `
<div class="e-col order-sm-1 order-md-3 order-lg-4 order-xl-2">
</div>
`;

exports[`ECol should test "grid" and match snapshot 1`] = `
<div class="e-col sm-12 md-6 lg-4 xl-12">
</div>
`;
